### PR TITLE
Revise APIs

### PR DESCRIPTION
**Pull request by kfangw**
### Github Issue
#337

### Background
revise apis to support features; page, queries, hal

### CODE REVIEW

File: lib/network/api/transaction_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		query = strings.Replace(query, "{reverse}", "true", 1)
+		receivedBts := testFunction(query)
+		require.Equal(t, len(btList), len(receivedBts), "length is not same")
+
+		for i, bt := range receivedBts {
+			require.Equal(t, bt.Hash, btList[len(btList) - 1 - i].Hash, "hash is not same")
+		}
+	}
+
+	//TODO: cursor
```
Comment (kfangw) [inline lib/network/api/transaction_test.go]: for the cursor, we need the response format to parse. @anarcher

Comment (anarcher) [reply lib/network/api/transaction_test.go]: I am doing to apply HAL to handlers #363 :->

File: lib/network/api/transaction_test.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+			bt := btMap[recv["hash"].(string)]
+			r := resource.NewTransaction(&bt)
+			txS, err := json.Marshal(r.Resource())
+			require.Nil(t, err)
+			require.Equal(t, txS, line)
+		}
+	}
+}
+
+func TestGetTransactionsHandlerPage(t *testing.T) {
```
Comment (kfangw) [inline lib/network/api/transaction_test.go]: good point to start the review

File: lib/network/api/stream.go [hunk 3]
```diff
@@ -47,6 +47,9 @@ var RenderJSONFunc = func(args ...interface{}) ([]byte, error) {
 		return nil, fmt.Errorf("render: value is empty") //TODO(anarcher): Error type
 	}
 	v := args[1]
+	if v == nil {
+		return nil, nil
+	}
```
Comment (anarcher) [inline lib/network/api/stream.go]: Thanks! @kfangw :+1:

**Comment by kfangw** [comment]
please wait for one more commit ^^; @anarcher

**Comment by kfangw** [comment]
Wrapping error with PROBLEM will be covered by next PR.

**Pull request merged**